lib/ext/ts.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+var fs = require('fs');
+
+var typescript;
+try {
+	typescript = require('typescript');
+} catch (err) {
+	console.log('AVA error: `--extensions=ts` require TypeScript to be installed.');
+	throw err;
+}
+
+var typescriptOptions = {
+	module: typescript.ModuleKind.CommonJS,
+	target: typescript.ScriptTarget.ES6,
+	sourceMap: true,
+	inlineSources: true,
+	inlineSourceMap: true
+};
+
+/**
+ *
+ * Ts-Node version
+ *
+ */
+function tsnodeRegister() {
+	try {
+		require('ts-node').register(typescriptOptions);
+	} catch (err) {
+		console.log('AVA error: `--extensions=ts` require ts-node to be installed.');
+		throw err;
+	}
+}
+
+function tsnodeTranspile(code, fileName) {
+	try {
+		const tsService = require('ts-node').register(typescriptOptions);
+		const result = tsService().compile(code, fileName);
+		return result;
+	} catch (err) {
+		console.log('AVA error: `--extensions=ts` require ts-node to be installed.');
+		throw err;
+	}
+}
+
+/**
+ *
+ * Typescript version
+ *
+ */
+function typescriptRegister() {
+	require.extensions['.ts'] = function (module, fileName) {
+		var code = fs.readFileSync(fileName, 'utf8');
+		var result = typescript.transpileModule(code, {
+			fileName,
+			compilerOptions: typescriptOptions,
+			reportDiagnostics: true
+		});
+		module._compile(result.outputText, fileName);
+	};
+
+	require('source-map-support').install({
+		hookRequire: true
+	});
+}
```

Comment:
Hi @novemberborn ,

I had simplified this module, removed all the ts-node parts, cleaned the code, but not just the transpile.

The `register()` function has to be kept because if we run test written in TypeScript, the lib/test-worker.js must has the ability to transpile TypeScript, which is enabled by `extTs.register()` at here: https://github.com/zixia/ava/blob/typescript/lib/test-worker.js#L38